rest_framework_json_api/renderers.py

```diff
@@ -382,6 +381,9 @@ def extract_root_meta(serializer, resource):
 
     @staticmethod
     def build_json_resource_obj(fields, resource, resource_instance, resource_name):
+        # Determine type from the instance if the underlying model is polymorphic
+        if isinstance(resource_instance, utils.POLYMORPHIC_ANCESTORS):
+            resource_name = utils.get_resource_type_from_instance(resource_instance)
```

Comment:
But doesn't this just work because Django Polymorphic's manager will resolve to the correct type? I don't see why we need code in the renderer when the types are already resolved to the correct model when using that manager...